src/convertToObject.js

```diff
@@ -3,16 +3,34 @@
 /**
  * Implement convertToObject function:
  *
- * Function takes a string with styles (see an example in [stylesString.js](./stylesString.js))
+ * Function takes a string with styles (see an example
+ * in [stylesString.js](./stylesString.js))
  * and returns an object where CSS properties are keys
- * and values are the values of related CSS properties (see an exampl in [test file](./convertToObject.test.js))
+ * and values are the values of related CSS properties
+ * (see an exampl in [test file](./convertToObject.test.js))
  *
  * @param {string} sourceString
  *
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const stylesArray = sourceString
+    .split(';')
+    .map(string => string.trim())
+    .filter(string => string);
+
+  const stylesObject = stylesArray.reduce((acc, string) => {
+    const [propertyName, propertyValue] = string
+      .split(':')
+      .map(stringPart => stringPart.trim());
+
+    return {
+      ...acc,
+      [propertyName]: propertyValue,
+    };
```

Comment:
**This piece of code return `NEW object` and overwrite the old one every iteration. !**

```suggestion
    const [propertyName, propertyValue] = string.split(':')

    acc[key.trim()] = value.trim();
   
    return acc;
```